### DISCUSSION TITLE
Use custom keybinding.

### DISCUSSION THREAD

**Discussion by shivanshu-semwal** [davatorium/rofi#1498] [Q&A] [answered]
I want to make a script where I get a menu with some options, and when I press the `Alt+1` keybind I get some value in return, and `Alt+2` will give some another value in return.

```bash
#!/bin/bash
CHOICE=$(echo "1\n2\n3\n" | rofi -dmenu -p "select")

# if i press Alt+1 choice should be - 0, and on pressing alt+2 it should be 4
# now i will perform some actions based on choice value
```

How can I make such script, I read [https://github.com/davatorium/rofi-scripts](https://github.com/davatorium/rofi-scripts) but they don't use such methods.

Is there a way to make such script?

Thanks.

**Comment by DaveDavenport** [comment, marked as answer]
See the `-kb-select-*`  keys in rofi. (run `rofi -show keys`) for how they are currently bound.

**Reply by shivanshu-semwal** [reply]
![image](https://user-images.githubusercontent.com/53912074/136665315-46a3d7e4-d9a5-459b-9fed-8240e442fd20.png)

It shows *Custom keybinding 1*  for `Alt+1`. 

> Key bindings can be modified using the configuration systems. Multiple keys can be bound to one action by comma separating them. For example -kb-primary-paste "Conctrol+v,Insert"

So I have to add `-kb-custom-1 "Alt+1,0"` ? Where is list of actions, i can't find them in man page.

**Reply by DaveDavenport** [reply]
I am not sure what you want. 
`kb-select-2`  select row  2 (so it returns the entry @ in row 2)

`kb-custom-2`   returns the current selected item and modifies the result value ($?) of rofi to 11 (custom-3 is 12, etc.).

**Reply by shivanshu-semwal** [reply]
> modifies the result value ($?) of rofi to 11 (custom-3 is 12, etc.).

Thanks this is what I wanted. Now I can check the  return value to know which custom keybind was pressed.

**Reply by DaveDavenport** [reply]
great!

**Comment by kamalarieff** [comment]
Hey @shivanshu-semwal, were you able to do this? When I press `kb-custom-2`, it doesn't return 11.

**Reply by shivanshu-semwal** [reply]
Here is how I did it.
```bash
#!/bin/bash
CHOICE=$(echo "1\n2\n3\n" | rofi -dmenu -p "select")

# if i press Alt+1 choice should be - 0, and on pressing alt+2 it should be 4

# check error code first to see which custom keybind was pressed 
TEMP="$?" # store the error code first
if [ $TEMP -eq 10 ]; then
    CHOICE="0" # alt + 1 (kb-custom-1)  was pressed
elif [ $TEMP -eq 11 ]; then
    CHOICE="4" # alt + 2 (kb-custom-2) was pressed
fi

# now do the action based on CHOICE
# ...
```

**Reply by kamalarieff** [reply]
Nice! Thank you for this.
